main.go

```diff
@@ -41,6 +43,16 @@ func main() {
 	// Register the service with the gRPC server
 	snapshotsapi.RegisterSnapshotsServer(rpc, service)
 
+	var gracefulstop = make(chan os.Signal)
+	signal.Notify(gracefulstop, syscall.SIGTERM)
+	signal.Notify(gracefulstop, syscall.SIGINT)
+	signal.Notify(gracefulstop, syscall.SIGSTOP)
+	go func() {
+		<-gracefulstop
+		rpc.GracefulStop()
```

Comment:
This looks good, but I would expand this by adding a timeout on the `GracefulStop()` call since it could block forever waiting for every gRPC clients to disconnect.
And if you reach the timeout, you could issue a `rpc.Stop()` which should force stop the server.

Reply:
@ganeshmaharaj will this cancel any pending operations. 

How does this map to the ctx you got passed in in your callbacks?

How can you cancel those operations?

Does rpc.GracefulStop invoke cancellations on the callbacks?

Reply:
@sboeuf from the (docs)[https://godoc.org/google.golang.org/grpc#Server.GracefulStop] i do not think gracefulstop accepts a timeout. Think it is good to run the timer outside?

Reply:
@mcastelino grpc's Gracefulstop will wait for all pending calls to complete before it stops the server and will not accept any new requests while it is in this state. for the operations that are already active and canceling those, we should implement the timeout within the context as we previously discussed.